File: fixture-app-node/order.test.js
const { summarize } = require("./order");

describe("summarize", () => {
  test("does not throw and returns a fallback when order has no customer", () => {
    const order = { amount: 42 };
    expect(() => summarize(order)).not.toThrow();
    expect(summarize(order)).toBe("Unknown owes 42");
  });

  test("uses the customer name when a customer is present", () => {
    const order = { customer: { name: "Ada" }, amount: 10 };
    expect(summarize(order)).toBe("Ada owes 10");
  });
});

File: fixture-app-go/order_test.go
package main

import "testing"

// TestSummarizeNilCustomer verifies that Summarize does not panic when the
// Order has no Customer attached. Before the fix, this caused:
// "panic: runtime error: invalid memory address or nil pointer dereference".
func TestSummarizeNilCustomer(t *testing.T) {
	o := &Order{Amount: 42}

	got := Summarize(o)

	want := "unknown customer owes 42"
	if got != want {
		t.Fatalf("Summarize(%+v) = %q, want %q", o, got, want)
	}
}

// TestSummarizeNilOrder verifies that Summarize does not panic when passed
// a nil *Order.
func TestSummarizeNilOrder(t *testing.T) {
	got := Summarize(nil)

	want := "no order"
	if got != want {
		t.Fatalf("Summarize(nil) = %q, want %q", got, want)
	}
}

// TestSummarizeWithCustomer verifies existing behavior still works when a
// Customer is present.
func TestSummarizeWithCustomer(t *testing.T) {
	o := &Order{Customer: &Customer{Name: "Alice"}, Amount: 10}

	got := Summarize(o)

	want := "Alice owes 10"
	if got != want {
		t.Fatalf("Summarize(%+v) = %q, want %q", o, got, want)
	}
}
